### PR TITLE
ms\applink.c: Option to suppress includes

### PR DESCRIPTION
**Pull request by ideafarm**
Define APPLINK_NOiNCLUDES to suppress includes.  This is useful when the symbols in applink.c must be modified in order to use wrappers.  The latter is needed when the compiler uses a default calling convention other than cdecl for its C library functions.  (E.g. Open Watcom.)

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: ms/applink.c [hunk 1]
```diff
@@ -34,9 +34,12 @@
 #define APPLINK_MAX     22      /* always same as last macro */
 
 #ifndef APPMACROS_ONLY
+
+#ifndef APPLINK_NOiNCLUDES
```
Comment (mattcaswell) [inline ms/applink.c:38]: The mixed case looks very strange to me. I would just call this `APPLINK_NO_INCLUDES`

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Comment by ideafarm** [comment]
Thank you, Mr. Caswell. Good catch.

------- Original Message -------

**Review by mattcaswell — APPROVED** [review]
Approved.

This PR has merge commits which we do not allow, but that can be resolved by whoever commits this on push to the repo.

**Comment by ideafarm** [comment]
I apologize for screwing this contribution up. It is my lifetime first opensource contribution and I don't really understand the process.

I understand you to be saying that someone else can fix this and that you do not need me to do anything else. Let me know if you need me to do something.
------- Original Message -------

**Comment by ideafarm** [comment]
Please reject PR #21556. I just deleted and reforked and redid the same changes and made a new PR: https://github.com/openssl/openssl/pull/21595.

------- Original Message -------

**Pull request closed without merging**